Silicon/NXP/iMXPlatformPkg/Drivers/SmbiosConfigDxe/SmbiosConfigDxe.c

```diff
@@ -0,0 +1,705 @@
+
+/**
+
+  Finds and retrieves the next Key/Value pair from the buffer starting at the
+  given offset.
+
+  This routine scans through the provided buffer starting at the given offset.
+  It finds the first '"' character and records its location as the start of the Key.
+  Then it continues scanning to find the next '"' instance. The length denotes the KeyLength.
+
+  We repeat these steps to find the Value position and length.
+
+  At any point if a failure is detected, the algorithm bails with an error.
+
+  If we successfully find both Key and Value positions and lengths, then allocate/copy
+  these to their respective buffers and return to the caller.
+
+**/
+EFI_STATUS
+EFIAPI
+GetNextKeyValuePair (
+  CHAR8 *Buffer,
+  UINTN *Offset,
+  UINTN BufferSize,
+  CHAR8 **Key,
+  CHAR8 **Value
+  )
+{
+  EFI_STATUS Status;
+  UINTN CurrentOffset;
+  UINTN KeyLength;
+  UINTN ValueLength;
+  JSON_PARSER_STATE State;
+  CHAR8 *KeyStart;
+  CHAR8 *ValueStart;
+  CHAR8 *TempKey;
+  CHAR8 *TempValue;
+
+  CurrentOffset = *Offset;
+  KeyLength = 0;
+  ValueLength = 0;
+  Status = EFI_NOT_FOUND;
+  State = StateFindKey;
+  KeyStart = NULL;
+  ValueStart = NULL;
+  TempKey = NULL;
+  TempValue = NULL;
+
+  while (CurrentOffset < BufferSize) {
+    if (State == StateValueDone) {
+      Status = EFI_SUCCESS;
+      break;
+    }
+
+    switch (State) {
+      case StateFindKey:
+        if (Buffer[CurrentOffset] == '"') {
+          State = StateReadKey;
+        }
+      break;
+      case StateReadKey:
+        if (KeyStart == NULL) {
+          KeyStart = &Buffer[CurrentOffset];
+        }
+        if (Buffer[CurrentOffset] == '"') {
+          State = StateKeyDone;
+        } else {
+          KeyLength++;
+        }
+      break;
+      case StateKeyDone:
+        if (Buffer[CurrentOffset] == ':') {
+          State = StateFindValue;
+        }
+      break;
+      case StateFindValue:
+        if (Buffer[CurrentOffset] == '"') {
+          State = StateReadValue;
+        }
+      break;
+      case StateReadValue:
+        if (ValueStart == NULL) {
+          ValueStart = &Buffer[CurrentOffset];
+        }
+        if (Buffer[CurrentOffset] == '"') {
+          State = StateValueDone;
+        } else {
+          ValueLength++;
+        }
+      break;
+      default:
+        DEBUG ((DEBUG_ERROR, "%a: Reached an unknown state %d\n", __FUNCTION__, State));
+        Status = EFI_NOT_FOUND;
+        goto Exit;
+    }
+    CurrentOffset++;
+  }
+
+  if (EFI_ERROR (Status)) {
+    goto Exit;
+  }
+
+  // Bounds check on Key and Value
+  if ((KeyLength == 0) || (ValueLength == 0)) {
+    Status = EFI_NOT_FOUND;
+    goto Exit;
+  }
+  if ((KeyLength > MAX_VARIABLE_SIZE) || (ValueLength > MAX_VARIABLE_SIZE)) {
+    Status = EFI_INVALID_PARAMETER;
+    goto Exit;
+  }
+  
+  TempKey = AllocatePool(KeyLength + 1);
+  if (TempKey == NULL) {
+    Status = EFI_OUT_OF_RESOURCES;
+    goto Exit;
+  }
+  CopyMem(TempKey, KeyStart, KeyLength);
```

Comment:
Add a space after function name. There are several more instances throughout the file.

Reply:
Gotta point them out since I already did a scrub and missed them